### ISSUE TITLE
Select multiple tabs for reordering

### ISSUE THREAD

**Issue by JohannLange** [microsoft/vscode#119327]
Whenever I'm reordering my tabs and moving them between the panes, I have to select each tab individually.
It would be awesome, if I could select multiple tabs at once to reorder them or move them between panes.
Basically similar to the behavior of every modern browser, where `Shift + click` selects a range of tabs or `Ctrl + click` for selected individual tabs.

**Comment by bpasero** [comment]
Yeah I was waiting for this feature request, I think it would even be cool to allow to select multiple tabs and allow to perform actions on all of them from the context menu.

**Comment by JohannLange** [comment]
> Yeah I was waiting for this feature request, I think it would even be cool to allow to select multiple tabs and allow to perform actions on all of them from the context menu.

I didn't even think about all the other possible actions, good catch!

**Comment by NatoBoram** [comment]
In Firefox, I often <kbd>Shift</kbd>+<kbd>Click</kbd> to select multiple tabs and <kbd>Ctrl</kbd>+<kbd>W</kbd> to close them. I find myself doing this in VSCode multiple times per day even though it doesn't do anything.

**Comment by ccarner** [comment]
> In Firefox, I often Shift+Click to select multiple tabs and Ctrl+W to close them after some intense Googling and StackOverflowing. I find myself doing this in VSCode multiple times per day even though it doesn't do anything.

I did this just before, for the 100th time today haha. I'd love this change :)

**Comment by Binly42** [comment]
Another point similar to modern browsers: Allow to move **a whole group even with too many tabs**.

If there are too many tabs in a group, I cannot drag the group because there is **no available area** upon where the "mouse arrow cursor" would change to be a "hand cursor" ......

---

Excuse me If this should be a new issue, but I guess this is not a big issue and is also relative to "modern browser feature"...

Since in my opinion, there should just might be somewhat relationship between:
- operations on multiple tabs (even of one more groups?)
- operations on all tabs of a group (which are some existed features)

**Comment by Tyriar** [comment]
I actually tried to do this a few times over the weekend instinctively as a way to close many editors at a time as the alternative is to hit ctrl+w 20 times, and ctrl+shift+t to undo when I closed too many.

**Comment by Kenobi-the-2nd** [comment]
Yes, I have been looking for this feature for some time. It would be very helpful for organizational purposes, especially since I typically work in multiple languages at once.

**Comment by RocketRace** [comment]
This is definitely useful in e.g. large codebases where a refactoring may touch dozens of files at once. I keep stumbling over my muscle memory! 😄

**Comment by stronglify13** [comment]
Wait for this!! :D

**Comment by RobinDavidss0n** [comment]
Any updates?

**Comment by starball5** [comment]
Is this implemented already? I can hold the alt key, click into the unoccupied space in an editor group's tab strip, and then drag and drop the tabs to other editor groups. It just doesn't seem to currently support creating new editor groups while dragging- only moving all tabs from one editor group into another.

Related on Stack Overflow: [How can I drag and drop editor groups to customize the layout in VS Code?](https://stackoverflow.com/q/76558769/11107541)

**Comment by narcis-fv** [comment]
Same here, would like to select multiple tabs, right click and click 'close others'

**Comment by pentago** [comment]
Any chance we can get this basic thing?

**Comment by starball5** [comment]
@pentago (tongue in cheek) any chance you can PR a draft implementation of this "basic" thing?

**Comment by pentago** [comment]
@starball5 not really. Im not saying it's a basic to code rather basic UX thing to have in tabbed windows, I'm sure you'll agree.

Also, this feature was present before, I remember it worked some years ago, not sure what happened to it.

**Comment by mkumarpcln** [comment]
+1, would be great to have!

**Comment by sundeep-ravella** [comment]
I will work on this

**Comment by Tyriar** [comment]
@sundeep-ravella I suspect this issue will be too complex for an external contribution.

**Comment by benibenj** [comment]
Tabs multi select is now supported in our insiders build. Give it a try and let us know what you think.

You can get VSCode Insiders here: https://code.visualstudio.com/insiders/

**Comment by pentago** [comment]
Just tested it, selection and movement works great. Thank you!
